### DISCUSSION TITLE
List of text to images with just that text on them.

### DISCUSSION THREAD

**Discussion by aceandduce** [ImageMagick/ImageMagick#4768] [Help]
I have been struggling to figure out how to do this and ImageMagick came up as a possible solution. To be more clear I am trying to convert a list of words (or short phrases) into individual images for each part of the list. 

Here is an example of what I mean.
List;
Object
Object One
Thing
Stuff

Each line of the list would be put atop it's own image, the hope being it wraps properly for the space limitations given (ex a 256x256 black square as the base image).

I would greatly appreciate the help and I want to thank you just for taking the moment to read this.

**Comment by fmw42** [comment]
See caption:  https://legacy.imagemagick.org/Usage/text/#caption.  If more than one word, it will wrap to next line.  

convert -background black -fill white -font Arial -size 256x256 -gravity north caption:"some_phrase" result.jpg

![result](https://user-images.githubusercontent.com/13458218/151298572-725de137-db29-4117-a5f8-b152c78bf64b.jpg)

or use label: for one or more words, https://legacy.imagemagick.org/Usage/text/#label

convert -background black -fill white -font Arial -size 256x256 -gravity north label:"some phrase" result2.jpg

![result2](https://user-images.githubusercontent.com/13458218/151298851-4b79d6f8-2663-4fb5-8eb9-26a103babac4.jpg)

**Reply by aceandduce** [reply]
THANK YOU SO MUCH! Now my only question is how can I go about going through a large list automatically? As in, it changes out the label or caption and uses the next line in the list after making the image of the previous one.

**Comment by snibgo** [comment]
See the recent thread https://github.com/ImageMagick/ImageMagick/discussions/4737

You need to say what version of IM you use, on what platform.

**Reply by aceandduce** [reply]
Thank you very much, that thread is seemingly what I need. However, I don't understand how to signify the actual file location in the bat file for either the list of words nor the output. This is my first time using ImageMagick.

**Comment by snibgo** [comment]
You need to say what version of IM you use, on what platform.

I showed a Windows BAT script:
```
for /F "tokens=*" %%L in (textfile.txt) do magick ^
  -pointsize 50 ^
  -background None ^
  caption:"%%L" ^
  "%%L.png"
```
> However, I don't understand how to signify the actual file location in the bat file for either the list of words nor the output.

I assumed your text file is named "textfile.txt", and is in your current directory. If it is named something else, replace that with the actual filename. If it isn't in the current directory, then use the full path to the file.

In my example, each output file is named after the corresponding text line, with a ".png" extension. You can change "%%L.png" to anything you like.

**Reply by aceandduce** [reply]
Thank you very much this helped me along my way greatly, now I have one final question. How do I go about having the text wrap if a sufficiently long word hits the bounds of the image? Currently, the word will spill over no matter what, even when doing what I believe should auto-resize it.

**Reply by fmw42** [reply]
See label: or caption: at the links I recommended above.
